frontend/controller/actions/identity.js

```diff
@@ -406,7 +406,7 @@ export default (sbp('sbp/selectors/register', {
       }
 
       try {
-        if (!state) {
+        if (!cheloniaState?.contracts[identityContractID]?.references) {
```

Comment:
It was at least somewhat followable before, but now this line seems to come out of nowhere.

Could you please add a detailed comment explaining why we're checking this instead of `state`?

Reply:
This is what fixes the second bug. I'll add a comment explaining.

Reply:
Would this comment look OK?

```
// When syncing from scratch, we need to call `retain` on our own identity
// contract to ensure the reference count is positive and we don't accidentally
// remove it.
// When not syncing from scratch, we call `sync` instead, since we don't want to
// increment the reference count every time the app is loaded.
// To decide what to call, we check if the reference count exists. This check
// is correct and also ensures that, if the state gets corrupted for some reason
// (for example, because the contract is deleted from the server and then
// restored) and the reference count lost, we call `/retain` again.
```